pkg/utils/networkinggadgets.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"os"
+
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	containerutils "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils"
+)
+
+type NetworkingTracer[Event any] interface {
+	Attach(key string, pid uint32, eventCallback func(Event)) error
+	Detach(key string) error
+}
+
+func ConnectTracerToContainerCollection[Event any](
```

Comment:
Suggestion:

```suggestion
func ConnectNetworkingTracerToContainerCollection[Event any](
```

Since it's doing specific depending on the network namespace used.